run_bamstrap

```diff
@@ -12,22 +11,48 @@ if [ "$(id -u)" = 0 ]; then
   exit 4
 fi
 
-if ! command -v curl >/dev/null 2>&1; then
-  echo "Aborting! No curl found (install and retry)"
-  exit 5
+echo "Requesting sudo access to install packages if needed"
+sudo date
+
+if ! [ -x "$(command -v curl)" ]; then
+  echo "Installing curl..."
+  sudo apt-get install -y curl
 fi
 
-if ! command -v ruby >/dev/null || \
-   (command -v ruby >/dev/null && ruby -e 'exit RUBY_VERSION.split(".").first(2).join(".").to_f < 2.3'); then
-  echo "No suitable Ruby found (installing vendor Ruby)"
-  eval "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install-ruby)"
+if [ ! -f "$HOME/.ssh/id_rsa.pub" ]; then
+  echo "Ensure you have your GitHub SSH key at ~/.ssh/id_rsa.pub"
```

Comment:
Looks like (at least) GitHub recommend a `ed25519 ` key, which doesn't use the same naming convention:
https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#generating-a-new-ssh-key

> If you are using a legacy system that doesn't support the Ed25519 algorithm, use: [an rsa key]